unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TransactionHUD/TransactionHUD.cs

```diff
@@ -0,0 +1,122 @@
+using System;
+using TMPro;
+using System.Collections.Generic;
+using DCL.Controllers;
+using DCL.Helpers;
+using UnityEngine;
+using UnityEngine.UI;
+using DCL.TransactionHUDModel;
+using UnityEngine.SocialPlatforms.Impl;
+using Type = DCL.TransactionHUDModel.Type;
+
+public class TransactionHUD : MonoBehaviour, ITransactionHUD
+{
+    [SerializeField] private GameObject paymentPanel;
+    
+    [SerializeField] private GameObject signPanel;
+
+    [SerializeField] private List<Button> acceptButtons;
+    
+    [SerializeField] private List<Button> rejectButtons;
+    
+    [SerializeField] private TMP_Text amountLabel;
+    
+    [SerializeField] private TMP_Text fromAddressLabel;
+
+    [SerializeField] private TMP_Text toAddressLabel;
+    
+    [SerializeField] private TMP_Text messageLabel;
+    
+    [SerializeField] private TMP_Text paymentPanelTitle;
+    
+    [SerializeField] private TMP_Text signPanelTitle;
+    
+    [SerializeField] private TMP_Text paymentNetworkLabel;
+    
+    [SerializeField] private TMP_Text signNetworkLabel;
+
+    public Model model { get; private set; } = new Model();
+
+    public event Action<ITransactionHUD> OnTransactionAccepted;
+    
+    public event Action<ITransactionHUD> OnTransactionRejected;
+
+    private void OnEnable()
+    {
+        foreach (var acceptButton in acceptButtons)
+            acceptButton.onClick.AddListener(AcceptTransaction);
+        
+        foreach (var rejectButton in rejectButtons)
+            rejectButton.onClick.AddListener(RejectTransaction);
+    }
+    
+    public IParcelScene FindScene(string sceneId)
+    {
+        foreach (IParcelScene scene in DCL.Environment.i.world.state.scenesSortedByDistance)
+        {
+            if (scene.sceneData.id == sceneId)
+                return scene;
+        }
+
+        return null;
+    }
+
+    private void OnDisable()
+    {
+        foreach (var acceptButton in acceptButtons)
+            acceptButton.onClick.RemoveAllListeners();
+        
+        foreach (var rejectButton in rejectButtons)
+            rejectButton.onClick.RemoveAllListeners();
+    }
+
+    private static string ShortAddress(string address)
+    {
+        return $"{address.Substring(0, 6)}...{address.Substring(address.Length - 4)}";
+    }
+
+    public void Show(Model model)
+    {
+        if (Utils.IsCursorLocked)
+            Utils.UnlockCursor();
+
+        this.model = model;
+
+        var scene = FindScene(model.sceneId);
+
+        if (model.requestType == Type.REQUIRE_PAYMENT)
```

Comment:
You can extract both blocks of code of this `if` to add more descriptive methods to reduce complexity, like `ShowTransferMessage` and `ShowSignMessage`.